components/Tag.js

```diff
@@ -1,17 +1,62 @@
 import Link from "../components/Link";
 import { abbreviateNumber } from "../services/utils/abbreviateNumbers";
 
-export default function Tag({ name, total }) {
+export default function Tag({
+  name,
+  total,
+  path,
+  currentInput,
+  selected,
+  setInputValue,
+}) {
   return (
     <div className="relative">
       <Link
-        href={`/search?search=${name}`}
-        className="flex flex-row p-1 m-2 rounded-lg text-sm font-mono border-2 cursor-pointer shadow-none hover:border-orange-600"
+        onClick={() => {
+          if (!currentInput) {
+            return;
+          }
+          if (!selected) {
```

Comment:
Replace the whole part with this please. 

```
if (!selected) {
            if (currentInput !== "") {
              setInputValue(
                `${currentInput
                  .toLowerCase()
                  .split(",")
                  .filter((input) => input !== "")
                  .join(",")},${name}`
              );
            }

            else {
              setInputValue(`${name}`);
            }
          }

          else{
            setInputValue(
              currentInput
                .toLowerCase()
                .split(",")
                .filter((input) => input !== "" && input !== name.toLowerCase())
                .join(",")
            );
          }
```

Reply:
@eddiejaoude I checked the code and the tags on the search page were not getting enabled directly. So I tweaked a bit and then it worked fine. Please make the above 2 changes that I have asked.

Reply:
Thank you @Ananya2001-an for being super productive with this feature. I tweaked it further to simplify the code as it was quiet complicated, you had almost built a search engine 😂 

We will add a search engine later, so I was keen to keep this simple, have a look and let me know what you think (java and javascript are still both triggered, but I was thinking, where do we stop, there are many other namings that will also clash and I don't want to manage a whole dictionary of endless possibilities)

I look forward to hearing your feedback, thanks again

Reply:
@eddiejaoude I really did overthink everything and actually created a search engine 😂
But I agree with you on the fact to keep it simple. I tested it out right now and it works great just a few things I want you to change if that's possible:

1. In the selected condition make the tag name lowercase so that it gets highlighted no matter the input otherwise user has to keep in mind the exact case.

2. Remove tagNames array in the search.js file it's no longer being used

3. If possible for the selected condition you can convert the inputValue into an array with all values in lowercase and then check if the tag.name.toLowercase() is present or not. This will remove the error of java and react getting highlighted when clicking on javascript and reactjs respectively.

4. One last thing that on the user profile the links are now blue in colour you need to adjust the css there.

That's it I guess. Overall I am really happy with the simplified working ❤️

Reply:
Also it was right to remove the tags from the discover page. It was becoming quite redundant.

Reply:
@Ananya2001-an great points 💯 

1. ✅ Good point
2. ✅ Thank you, good spot
3. Sounds like a great idea, but not sure I follow how to do it. Could you create a new PR for this?
4. ✅ Good spot

Reply:
Sure I will raise a new PR to fix the 3rd issue. Besides that congrats on the new feature 👏

Reply:
Also in the same PR I will make a change for the Home page where we need to update the Search page screenshot

Reply:
Great 👍  Thank you 🚀